### ISSUE TITLE
[2.15.6] Regression in rev-parse deploying git commit

### ISSUE THREAD

**Issue by felixbuenemann** [capistrano/capistrano#1467]
The changes introduced in 8294c5e break deployment of revisions that point to a commit instead of a branch:

```
executing locally: "git rev-parse --revs-only origin/ad9f3e9"
# returns nothing
/Users/felix/.gem/ruby/2.2.0/gems/capistrano-2.15.6/lib/capistrano/recipes/deploy/scm/git.rb:245:in `query_revision': Unable to resolve revision for 'ad9f3e9' on repository '.'. (RuntimeError)
```

Reason:
```sh
# old behavior:
git rev-parse --revs-only ad9f3e9
ad9f3e90008c829727d7a2c89d557a0ba3ad179e
# new behavior:
git rev-parse --revs-only origin/ad9f3e9
# no output
```

**Comment by joelataylor** [comment]
@felixbuenemann - do you think this regression is also showing itself with branch deploys?

`BRANCH_NAME=homepage-menu-type-info cap staging deploy `

results in:
```
Tasks: TOP => git:set_current_revision
(See full trace by running task with --trace)
The deploy has failed with an error: Exception while executing as oamm@0.0.0.100: git exit status: 128
git stdout: Nothing written
git stderr: fatal: ambiguous argument 'homepage-menu-type-info': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions
```

If I remove the remote `repo` directory, and rerun, it works.

**Comment by felixbuenemann** [comment]
Hard to say without the git command that was executed.

**Comment by joelataylor** [comment]
@felixbuenemann - sorry, here ya go:
`[142f90a9] Running /usr/bin/env git rev-list --max-count=1 --abbrev-commit master as oamm@100.100.100.100`

**Comment by felixbuenemann** [comment]
That looks like a different problem.

**Comment by joelataylor** [comment]
Yeh - I tracked it down to a SubmoduleStrategy class that was causing the problem. :) Sorry for my intrusion on your issue.

**Comment by mattbrictson** [comment]
Can this be closed now that #1470 has been merged?

**Comment by felixbuenemann** [comment]
Yes, it's fixed in 2.15.7, just forgot to close the issue.
